lib/pbench/agent/results.py

```diff
@@ -117,13 +127,14 @@ def make_result_tb(self):
             except OSError as exc:
                 if exc.errno != errno.ENOENT:
                     self.logger.error("error removing failed tar ball, {}", tarball)
-            sys.exit(1)
+            return (1, "")
 
-        self.make_md5sum(tarball)
+        ret = self.make_md5sum(tarball)
+        return (ret, "")
 
         # The contract with the caller is to just return the full path to the
         # created tar ball.
-        return str(tarball)
+        return (0, str(tarball))
```

Comment:
I'm sure that you're already hip-deep in fixing this, but I just noticed that line 137 is _unreachable_ due to the unconditional `return` statement at line 133.

Hopefully your solution to the error reporting mechanism for this function will rectify this.  :-)

Reply:
yeah working on that